### PR TITLE
Add unit test for GCP integration module

### PR DESCRIPTION
**Pull request by Molter73**
|Related issue|
|---|
|#4015|

## Description
This PR adds unit testing for the gcp integration module.

## Tests
|Function|Status|
|---|---|
|wm_gcp_run|Done|
|wm_gcp_destroy|Done|
|wm_gcp_dump|Done|
|wm_gcp_main|Done|
|wm_gcp_read|Done|

### CODE REVIEW

**Comment by Molter73** [comment]
### Tests on wm_gcp_run
|Test|Result|Comments|
|---|---|---|
|test_wm_gcp_run_success_log_disabled|Pass|Every execution with an output from the script writes "Logging disabled" into the log.|
|test_wm_gcp_run_error_running_command|Pass||
|test_wm_gcp_run_unknown_error|Pass||
|test_wm_gcp_run_unknown_error_no_description|Pass||
|test_wm_gcp_run_error_parsing_args|**Fail**|Output on mtwarn has a wrong format|
|test_wm_gcp_run_error_parsing_args_no_description|Pass||
|test_wm_gcp_run_generic_error|Pass||
|test_wm_gcp_run_logging_debug_message_debug|Pass||
|test_wm_gcp_run_logging_debug_message_not_debug|Pass||
|test_wm_gcp_run_logging_debug_message_not_debug_discarded|Pass||
|test_wm_gcp_run_logging_info_message_info|Pass||
|test_wm_gcp_run_logging_info_message_debug|Pass||
|test_wm_gcp_run_logging_info_message_warning|**Fail**|Messages with a severity higher than INFO are not written to the log|
|test_wm_gcp_run_logging_warning_message_warning|Pass||
|test_wm_gcp_run_logging_warning_message_debug|Pass||
|test_wm_gcp_run_logging_warning_message_error|**Fail**|Messages with a severity higher than WARNING are not written to the log|
|test_wm_gcp_run_logging_error_message_error|Pass||
|test_wm_gcp_run_logging_error_message_info|Pass||
|test_wm_gcp_run_logging_error_message_critical|**Fail**|Messages with a severity higher than ERROR are not written to the log|
|test_wm_gcp_run_logging_critical_message_critical|Pass||
|test_wm_gcp_run_logging_critical_message_debug|Pass||

**Comment by Molter73** [comment]
### Tests on wm_gcp_dump
|Test|Result|Comments|
|--|--|--|
|test_wm_gcp_dump_success_logging_disabled|Pass||
|test_wm_gcp_dump_success_logging_debug|Pass||
|test_wm_gcp_dump_success_logging_info|Pass||
|test_wm_gcp_dump_success_logging_warning|Pass||
|test_wm_gcp_dump_success_logging_error|Pass||
|test_wm_gcp_dump_success_logging_critical|Pass||
|test_wm_gcp_dump_success_logging_default|Pass||
|test_wm_gcp_dump_error_allocating_wm_wd|Pass||
|test_wm_gcp_dump_error_allocating_root|Pass||

**Comment by Molter73** [comment]
### Tests on wm_gcp_destroy
|Test|Result|Comments|
|--|--|--|
|test_wm_gcp_destroy|Pass|No assertions are possible on this test, however, running it under valgrind shows a memory leak in it.|

**Comment by Molter73** [comment]
### Tests on wm_gcp_main
|Test|Result|Comments|
|--|--|--|
|test_wm_gcp_main_disabled|Pass||
|test_wm_gcp_main_pull_on_start|Pass||
|test_wm_gcp_main_wait_before_pull|Pass||

**Comment by Molter73** [comment]
### Tests on wm_gcp_read
|Test|Result|Comments|
|---|---|---|
|test_wm_gcp_read_full_configuration|Pass||
|test_wm_gcp_read_enabled_tag_invalid|Pass||
|test_wm_gcp_read_project_id_tag_invalid|Pass||
|test_wm_gcp_read_no_project_id_tag|Pass||
|test_wm_gcp_read_subscription_name_tag_invalid|Pass||
|test_wm_gcp_read_no_subscription_name_tag|Pass||
|test_wm_gcp_read_credentials_file_full_path|Pass||
|test_wm_gcp_read_credentials_file_tag_empty|Pass||
|test_wm_gcp_read_credentials_file_tag_too_long|Pass||
|test_wm_gcp_read_credentials_file_tag_realpath_error|Pass||
|test_wm_gcp_read_credentials_file_tag_file_not_found|Pass||
|test_wm_gcp_read_no_credentials_file_tag|Pass||
|test_wm_gcp_read_max_messages_tag_empty|**Fail**|This test fails because the way of checking if the max_messages tag has content is wrong.|
|test_wm_gcp_read_max_messages_tag_not_digit|Pass||
|test_wm_gcp_read_pull_on_start_tag_invalid|Pass||
|test_wm_gcp_read_logging_tag_debug|Pass||
|test_wm_gcp_read_logging_tag_info|Pass||
|test_wm_gcp_read_logging_tag_warning|Pass||
|test_wm_gcp_read_logging_tag_error|Pass||
|test_wm_gcp_read_logging_tag_critical|Pass||
|test_wm_gcp_read_logging_tag_invalid|Pass||
|test_wm_gcp_read_invalid_tag|Pass||
|test_wm_gcp_read_invalid_element|Pass||

File: src/unit_tests/wazuh_modules/test_wmodules_gcp.c [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+    check_expected_ptr(nodes);
+    check_expected(MODULE_NAME);
+
+    return mock();
+}
+
+char *__wrap_realpath(const char *path, char *resolved_path) {
+    check_expected(path);
+
+    snprintf(resolved_path, PATH_MAX, mock_type(char*));
```
[the remark this answers is not in the document]
Comment (Molter73) [reply src/unit_tests/wazuh_modules/test_wmodules_gcp.c]: Fixed

**Comment by wazuhci** [comment]
The stress test has been launched. The charts will be posted when they are ready.
I will use the following configuration:
### General
- Commit: 3843d8f
- Packages: 
  - [manager 3.12.0 0.commit3843d8f rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-manager-3.12.0-0.commit3843d8f.x86_64.rpm)
  - [agent 3.12.0 0.commit3843d8f rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-agent-3.12.0-0.commit3843d8f.x86_64.rpm)
  - [manager 3.12.0 0.commit3843d8f deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-manager_3.12.0-0.commit3843d8f_amd64.deb)
  - [agent 3.12.0 0.commit3843d8f deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-agent_3.12.0-0.commit3843d8f_amd64.deb)
  - [agent 3.12.0 0.commit3843d8f msi](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/windows/wazuh-agent-3.12.0-0.commit3843d8f.msi)

### Stress test parameters:
- Manager system: `centos`
- Launch CentOS agent: `true`
- Launch Ubuntu agent: `true`
- Launch Windows agent: `true`
- Test duration: `60`
- Scan frequency: `10`
- Modules to test: 
  - logcollector: `true`
  - syscheck: `true`
  - rootcheck: `true`
  - sca: `true`
  - osquery: `true`
  - syscollector: `true`
  - active-response: `true`
  - azure-logs: `true`
  - cis-cat: `true`
  - docker-listener: `true`
  - open-scap: `true`
  - vulnerability-detector: `false`
- FIM special test: `None`
- FIM sleep: `300`
- FIM max files: `25000`
- SCA sleep: `30`
- SCA max registers: `5000`
- Docker image name: `hello-world`
- Docker sleep: `5`
- Docker jobs: `10`
- Logcollector special test: `None`
- Logcollector max files: `50000`
- Logcollector creation sleep: `300`
- Logcollector write files: `1000`
- Logcollector write sleep: `10`
- Vuln sleep: `60`
- Authd port: `1515`
- Syscollector package: `imagemagick.app`
- Syscollector package version: `7.0.8.48`

Regards!.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 3843d8f4690ce0ffcb2acfc6cee516254a4dd506
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B943/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B943/ossec.log)
**Coverage:** 

```json
Lines executed:30.87% of 1814

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B943/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.12.0-B943.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B943/Analysis_dynamic_valgrind-v3.12.0-B943.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 3843d8f4690ce0ffcb2acfc6cee516254a4dd506
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B944/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B944/ossec.log)
**Coverage:** 

```json
Lines executed:30.78% of 2339

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B944/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.12.0-B944.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B944/Analysis_dynamic_valgrind-v3.12.0-B944.log)

**Pull request merged**